src/webui/www/private/scripts/contextmenu.js

```diff
@@ -465,14 +477,25 @@ window.qBittorrent.ContextMenu ??= (() => {
 
             let first = true;
             for (const { categoryName, categoryHash } of sortedCategories) {
-                const el = new Element("li", {
-                    html: `<a href="javascript:torrentSetCategoryFN(${categoryHash});"><img src="images/view-categories.svg"/>${window.qBittorrent.Misc.escapeHtml(categoryName)}</a>`
+                const anchor = document.createElement("a");
+                anchor.href = `#Category/${categoryName}`;
+                anchor.textContent = categoryName;
+                anchor.addEventListener("click", (event) => {
+                    torrentSetCategoryFN(categoryHash);
                 });
```

Comment:
@Chocobo1, default action should be prevented here too, just like in updateTagsSubMenu

Reply:
After selecting a category item, the menu should close (due to only 1 category is allowed per torrent). I think the webui is correctly following the gui.
This is different to selecting a tag item which the menu shouldn't close (many tags is allowed per torrent).
Feel free to fix it if it wasn't the case.

Reply:
Event happens on an anchor so browser tries to follow up to a new URL:
![image](https://github.com/user-attachments/assets/2ca3fbd2-2678-4f76-9794-92becbb3ac6b)

Reply:
PR #21283.